spot-contracts/contracts/FeePolicy.sol

```diff
@@ -169,194 +173,124 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
-        uint256 currentVaultTVL = vault.getTVL();
-        IFeePolicy.SubscriptionState memory postBurnState = computeSubscriptionState(
-            perp.getDepositBond(),
-            currentPerpTVL - valueIn,
-            currentVaultTVL
-        );
-
-        // Split the fees between the perp and vault systems
-        uint256 vaultFeePerc = vaultSwapFeePerc[
-            keccak256(abi.encodePacked(address(perp), address(vault.underlying())))
-        ];
-        // When the system is under-subscribed, we charge no perp burn fee.
-        if (postBurnState.normalizedSubscriptionRatio <= 1) {
-            return (0, vaultFeePerc);
+    function perpToUnderlyingSwapFeePercs(uint256 sr) external override returns (uint256, uint256) {
+        // When the system is under-subscribed, perp share of fees is zero.
+        if (sr <= ONE) {
+            return (0, _vaultPerpToUnderlyingSwapFeePerc);
         }
 
-        // When the system is over-subscribed, we charge a perp burn fee on top of the vault's swap fee.
-        return (perpBurnFeePerc, vaultFeePerc);
+        return (_perpBurnFeePerc, _vaultPerpToUnderlyingSwapFeePerc);
     }
 
     /// @inheritdoc IFeePolicy
     function decimals() external pure override returns (uint8) {
         return DECIMALS;
     }
 
-    /// @notice Computes the detailed subscription state.
-    /// @return The subscription state.
-    function computeSubscriptionState() public override returns (IFeePolicy.SubscriptionState memory) {
-        return computeSubscriptionState(perp.getDepositBond(), perp.getTVL(), vault.getTVL());
-    }
-
-    /// @notice Computes the detailed subscription state based on the given `perpTVL` and `vaultTVL`.
-    /// @return The subscription state.
-    function computeSubscriptionState(
-        IBondController depositBond,
-        uint256 perpTVL,
-        uint256 vaultTVL
-    ) public view returns (IFeePolicy.SubscriptionState memory) {
-        IFeePolicy.SubscriptionState memory s;
-        s.perpTVL = perpTVL;
-        s.vaultTVL = vaultTVL;
-        // NOTE: We assume that perp only accepts the senior one.
-        //       We assume that perp's TVL and vault's TVL values have the same base denomination.
-        (, uint256 perpTR) = depositBond.tranches(0);
-        uint256 equilibriumVaultTVL = s.perpTVL.mulDiv(
-            TRANCHE_RATIO_GRANULARITY - perpTR,
-            perpTR,
-            MathUpgradeable.Rounding.Up
-        );
-        s.normalizedSubscriptionRatio = s.vaultTVL.mulDiv(ONE, equilibriumVaultTVL).mulDiv(
-            ONE,
-            targetSubscriptionRatio
-        );
-        return s;
+    /// @inheritdoc IFeePolicy
+    function computeSubscriptionRatio(IFeePolicy.SubscriptionParams memory s) public view returns (uint256) {
+        // NOTE: We assume that perp's TVL and vault's TVL values have the same base denomination.
+        return
+            s.vaultTVL.mulDiv(ONE, targetSubscriptionRatio).mulDiv(
```

Comment:
What about simply:
`(s.vaultTVL * s.perpTR) / (s.pertTVL * s.vaultTR);`

Or, if we want to keep the rounding up:
`s.vaultTVL.mulDiv(s.perpTR, s.pertTVL * s.vaultTR, MathUpgradeable.RoundingUp)`